core/src/test/java/org/apache/iceberg/rest/TestRESTScanPlanning.java

```diff
@@ -0,0 +1,871 @@
+              .first()
+              .as("Expected at least one task with delete files")
+              .actual();
+
+      assertThat(taskWithDeletes.file().location()).isEqualTo(FILE_A.location());
+      assertThat(taskWithDeletes.deletes()).hasSize(1); // 1 delete file: FILE_A_EQUALITY_DELETES
+      assertThat(taskWithDeletes.deletes().get(0).location())
+          .isEqualTo(FILE_A_EQUALITY_DELETES.location());
+    }
+  }
+
+  @ParameterizedTest
+  @EnumSource(PlanningMode.class)
+  void remoteScanPlanningWithMixedDeletes(
+      Function<TestPlanningBehavior.Builder, TestPlanningBehavior.Builder> planMode)
+      throws IOException {
+    configurePlanningBehavior(planMode);
+    Table table = restTableFor(scanPlanningCatalog(), "mixed_deletes_test");
+    setParserContext(table);
+
+    // Add both position and equality deletes in separate commits
+    table.newRowDelta().addDeletes(FILE_A_DELETES).commit(); // Position deletes for FILE_A
+    table
+        .newRowDelta()
+        .addDeletes(FILE_B_EQUALITY_DELETES)
+        .commit(); // Equality deletes for different partition
+
+    // Execute scan planning - should handle mixed delete types correctly
+    try (CloseableIterable<FileScanTask> iterable = table.newScan().planFiles()) {
+      List<FileScanTask> tasks = Lists.newArrayList(iterable);
+
+      // Verify task count: FILE_A only (FILE_B_EQUALITY_DELETES is in different partition)
+      assertThat(tasks).hasSize(1); // 1 data file: FILE_A
+
+      // Verify FILE_A with position deletes (FILE_B_EQUALITY_DELETES not associated since no
+      // FILE_B)
+      FileScanTask fileATask =
+          assertThat(tasks)
+              .filteredOn(task -> task.file().location().equals(FILE_A.location()))
+              .first()
+              .as("Expected FILE_A in scan tasks")
+              .actual();
+
+      assertThat(fileATask.deletes())
+          .hasSize(1); // 1 delete file: FILE_A_DELETES (FILE_B_EQUALITY_DELETES not matched)
+      assertThat(fileATask.deletes().get(0).location()).isEqualTo(FILE_A_DELETES.location());
+    }
+  }
+
+  @ParameterizedTest
+  @EnumSource(PlanningMode.class)
+  void remoteScanPlanningWithMultipleDeleteFiles(
+      Function<TestPlanningBehavior.Builder, TestPlanningBehavior.Builder> planMode)
+      throws IOException {
+    configurePlanningBehavior(planMode);
+    Table table = restTableFor(scanPlanningCatalog(), "multiple_deletes_test");
+    setParserContext(table);
+
+    // Add FILE_B and FILE_C to the table (FILE_A is already added during table creation)
+    table.newAppend().appendFile(FILE_B).appendFile(FILE_C).commit();
+
+    // Add multiple delete files corresponding to FILE_A, FILE_B, FILE_C
+    table
+        .newRowDelta()
+        .addDeletes(FILE_A_DELETES) // Position delete for FILE_A
+        .addDeletes(FILE_B_DELETES) // Position delete for FILE_B
+        .addDeletes(FILE_C_EQUALITY_DELETES) // Equality delete for FILE_C
```

Comment:
Minor: Could we go through some of these comments in the test and clean up the ones that are a bit redundant/obvious given the naming? It creates a bit more noise than needed.